File: bbarena-model/src/main/java/net/sf/bbarena/model/Referee.java
package net.sf.bbarena.model;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;
import java.util.Observable;

import net.sf.bbarena.model.team.Team;

public class Referee extends Observable implements Serializable {

	private static final long serialVersionUID = 5946955340981726094L;

	public enum RefereeStatus {
		NORMAL, BONDED, WATCHING;
	}

	private Map<Team, RefereeStatus> teamStatus = new HashMap<Team, RefereeStatus>();

	protected Referee(Team... teams) {
		for(Team team : teams) {
			teamStatus.put(team, RefereeStatus.NORMAL);
		}
	}

	public void setStatus(Team team, RefereeStatus status) {
		teamStatus.put(team, status);
		setChanged();
		notifyObservers();
	}

	public RefereeStatus getStatus(Team team) {
		return teamStatus.get(team);
	}

}

File: src/main/kotlin/bbarena/model/Referee.kt
package bbarena.model

import bbarena.model.team.Team
import java.io.Serializable
import java.util.*

data class Referee(val teams: List<Team>) : Observable(), Serializable {

    private val teamStatus = mutableMapOf<Team, RefereeStatus>()

    enum class RefereeStatus {
        NORMAL, BONDED, WATCHING
    }

    init {
        for (team in teams) {
            teamStatus[team] = RefereeStatus.NORMAL
        }
    }

    fun setStatus(team: Team, status: RefereeStatus) {
        teamStatus[team] = status
        setChanged()
        notifyObservers()
    }

    fun getStatus(team: Team): RefereeStatus? {
        return teamStatus[team]
    }

}
